src/sp_arbre_du_ghetto.c

```diff
@@ -0,0 +1,20 @@
+#include "php_snuffleupagus.h"
+
+void free_arbre_du_ghetto(arbre_du_ghetto *sapin) {
+  while (sapin) {
+    arbre_du_ghetto *tmp;
+    free(sapin->value);
+    free_arbre_du_ghetto(sapin->idx);
+    tmp = sapin;
+    sapin = sapin->next;
+    free(tmp);
+  }
+}
+
+arbre_du_ghetto *arbre_du_ghetto_new() {
+  arbre_du_ghetto *new = malloc(sizeof(arbre_du_ghetto));
```

Comment:
Please do use the `emalloc` thingy instead, or even better, `ecalloc`, since you want to set everything to zero. Do you even need a function for this btw?